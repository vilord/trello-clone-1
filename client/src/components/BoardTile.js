import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Icon } from 'semantic-ui-react';

import './BoardTile.css';

const BoardTile = ({ id, title, theme, favorite }) => (
  <Link
    to={`b/${id}`}
    className="BoardTile"
    style={{
      backgroundImage: theme.picture ? `url(${theme.picture})` : null,
      backgroundColor: !theme.picture ? theme.color : null,
    }}
  >
    <h4>{title}</h4>
    <Icon
      name="empty star"
      style={{
        color: favorite ? '#e6c60d' : 'white',
        right: favorite ? 0 : null,
        width: favorite ? '30px' : null,
      }}
    />
    <span className="BoardTile-fade" />
  </Link>
);

BoardTile.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  theme: PropTypes.object.isRequired,
  favorite: PropTypes.bool.isRequired,
};

export default BoardTile;
