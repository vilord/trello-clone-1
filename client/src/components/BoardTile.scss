.BoardTile {
  position: relative;
  display: flex;
  flex-direction: row;
  overflow: hidden;
  margin: 10px;
  max-width: 300px;
  width: 30%;
  height: 100px;
  border-radius: 3px;
  color: white;
  justify-content: space-between;

  &:hover {
    color: white;

    i {
      right: 0;
      width: 30px;
    }
  }

  h4 {
    z-index: 1;
    margin-top: 10px;
    margin-left: 10px;
    pointer-events: none;
  }

  i {
    position: absolute;
    right: -60px;
    z-index:1;
    padding-top:10px;
    padding-right: 10px;
    width: 90px;
    height: 40px;
    font-size: 14px;
    line-height: 1.5 !important;
    transition: all 0.1s;

    &:hover {
      font-size: 17px;
      line-height: 1.3 !important;
    }
  }
}

.BoardTile-fade {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.15);

  &:hover {
    background-color: rgba(0,0,0,0.25);
  }
}

