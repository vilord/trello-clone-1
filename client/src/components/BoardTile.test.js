import React from 'react';
import { shallow } from 'enzyme';

import BoardTile from './BoardTile';

describe('BoardTile Component', () => {
  beforeAll(() => {
    console.error = err => {
      throw new Error(err);
    };
  });

  let component;
  let minProps;
  beforeEach(() => {
    minProps = {
      id: 'y42LW46J9luq3Xq9XMly',
      title: 'Trello Clone',
      theme: {
        color: 'blue',
        picture: '',
      },
      favorite: false,
    };
    component = shallow(<BoardTile {...minProps} />);
  });

  it('renders', () => {
    expect(component).toHaveLength(1);
  });

  it('sets the Link address using the id', () => {
    expect(component.find('Link').prop('to')).toMatch(minProps.id);
  });

  it('sets the backgroundImage using props.theme.picture if available', () => {
    component.setProps({
      theme: {
        color: 'blue',
        picture: 'http://imgur.com/p/a13de',
      },
    });
    expect(component.find('Link').prop('style').backgroundImage).toMatch(
      minProps.theme.picture,
    );
  });

  it('sets the backgroundColor using props.theme.color if no picture', () => {
    expect(component.find('Link').prop('style').backgroundColor).toMatch(
      minProps.theme.color,
    );
  });

  it('sets changes the Icon star style using props.favorite', () => {
    const initStyle = component.find('Icon').prop('style');
    component.setProps({ favorite: true });
    expect(component.find('Icon').prop('style')).not.toEqual(initStyle);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
