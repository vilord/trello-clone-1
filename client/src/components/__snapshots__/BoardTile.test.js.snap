// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardTile Component matches snapshot 1`] = `
<Link
  className="BoardTile"
  replace={false}
  style={
    Object {
      "backgroundColor": "blue",
      "backgroundImage": null,
    }
  }
  to="b/y42LW46J9luq3Xq9XMly"
>
  <h4>
    Trello Clone
  </h4>
  <Icon
    as="i"
    name="empty star"
    style={
      Object {
        "color": "white",
        "right": null,
        "width": null,
      }
    }
  />
  <span
    className="BoardTile-fade"
  />
</Link>
`;
